apitools/base/py/credentials_lib.py

```diff
@@ -48,65 +48,57 @@ def SetCredentialsCacheFileLock(lock):
     cache_file_lock = lock
 
 
-# TODO(craigcitro): Expose the extra args here somewhere higher up,
-# possibly as flags in the generated CLI.
+# List of additional methods we use when attempting to construct
+# credentials. Users can register their own methods here, which we try
+# before the defaults.
+_CREDENTIALS_METHODS = []
```

Comment:
A similar approach (allowing arbitrary auth plugins) is used in boto.  One danger is that it makes backwards compatibility testing of top-level methods very difficult, as it may impact user-authored credential methods.

This doesn't block the PR, but it's something to be aware of.  If there are any compatibility guarantees (or disclaimers) that you want to call you, this would be a good place.